### ISSUE TITLE
FP for `expl_impl_clone_on_copy`

### ISSUE THREAD

**Issue by edwloef** [rust-lang/rust-clippy#14374]
### Summary

The lint seems to fire even in the presence of trait bounds on the `Copy` impl that aren't equal to trait bounds on the `Clone` impl, which as far as I can tell, should not be linted on.

### Lint Name

expl_impl_clone_on_copy

### Reproducer

```rs
pub struct Fp<T>(T);

impl<T> Copy for Fp<T> where T: Copy {}

impl<T> Clone for Fp<T>
where
    T: Clone,
{
    fn clone(&self) -> Self {
        Self(self.0.clone())
    }
}
```

```
warning: you are implementing `Clone` explicitly on a `Copy` type
   |
17 | / impl<T> Clone for Fp<T>
18 | | where
19 | |     T: Clone,
...  |
24 | | }
   | |_^
   |
note: consider deriving `Clone` or removing `Copy`
   |
17 | / impl<T> Clone for Fp<T>
18 | | where
19 | |     T: Clone,
...  |
24 | | }
   | |_^
```

### Version

```text
rustc 1.85.0 (4d91de4e4 2025-02-17)
binary: rustc
commit-hash: 4d91de4e48198da2e33413efdcd9cd2cc0c46688
commit-date: 2025-02-17
host: x86_64-unknown-linux-gnu
release: 1.85.0
LLVM version: 19.1.7
```

**Comment by profetia** [comment]
Deriving traits on types with generic args will add the necessary bounds, so I think linting here is actually reasonable.

**Comment by edwloef** [comment]
Ah yes, not sure why that didn't compile the first time I tried it. Thanks :)
